src/main/java/com/datastax/oss/protocol/internal/request/AuthResponse.java

```diff
@@ -51,10 +51,12 @@ public Codec(int protocolVersion) {
     public <B> void encode(B dest, Message message, PrimitiveCodec<B> encoder) {
       AuthResponse authResponse = (AuthResponse) message;
       ByteBuffer token = authResponse.token;
-      token.mark();
-      encoder.writeBytes(token, dest);
-      token.reset();
-      Bytes.erase(token);
+      if (token != null) {
```

Comment:
You still need to write the token if it is null:

```suggestion
      if (token == null) {
        encoder.writeBytes((ByteBuffer) null, dest);
      } else {
```

Reply:
huh, okay, thanks